bayesflow/coupling_networks.py

```diff
@@ -436,7 +439,7 @@ def _semantic_spline_parameters(self, parameters):
         """
 
         shape = tf.shape(parameters)
-        rank = len(shape)
+        rank = parameters.shape.rank
```

Comment:
tensorflow-specific, `len(shape)` or `parameters.ndim` should be fine. Safest would probably be `keras.ops.ndim`

Reply:
Changed `parameters.shape.rank` to use `keras.ops.ndim` in coupling_networks.py (see [47fbbbd](https://github.com/stefanradev93/BayesFlow/pull/165/commits/47fbbbd4acb22c391b483c845f73e6862094c07a))